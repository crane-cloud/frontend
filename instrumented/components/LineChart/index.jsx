import React from "react";
import PropTypes from "prop-types";
import {
  LineChart,
  Line,
  ResponsiveContainer,
  CartesianGrid,
  XAxis,
  YAxis,
  Label,
} from "recharts";
import "./LineChart.css";

const LineChartComponent = ({
  preview,
  data,
  yLabel,
  xLabel,
  xDataKey,
  yDataKey,
  lineDataKey,
}) => {
  if (preview) {
    return (
      <ResponsiveContainer className="GraphSection" width="100%" height="100%">
        {data && data.length > 0 ? (
        <LineChart data={data}>
          <Line
            dot={false}
            strokeWidth="2px"
            dataKey={lineDataKey}
            stroke="#008AC1"
          />
        </LineChart>
        ): (
          <div className="NoGraphData">Graph Data Unavailable</div>
        )}
      </ResponsiveContainer>
    );
  }

  return (
    <ResponsiveContainer className="GraphSection" width="100%" height="100%">
      { data && data.length > 0 ? (
        <LineChart data={data}>
          <CartesianGrid strokeDasharray="3" />
          <XAxis dataKey={xDataKey}>
            <Label
              className="ChartLabel"
              value={xLabel}
              position="centerBottom"
              dy={12}
            />
          </XAxis>
          <YAxis dataKey={yDataKey}>
            <Label
              className="ChartLabel"
              value={yLabel}
              angle={270}
              position="insideLeft"
              dy={-10}
            />
          </YAxis>
          
          <Line dot strokeWidth="2px" dataKey={lineDataKey} stroke="#008AC1" />
        </LineChart>
      ) : (
        <div className="NoGraphData">Graph Data Unavailable</div>
      )}
    </ResponsiveContainer>
  );
};

LineChartComponent.propTypes = {
  preview: PropTypes.bool,
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  yLabel: PropTypes.string,
  xLabel: PropTypes.string,
  xDataKey: PropTypes.string,
  yDataKey: PropTypes.string,
  lineDataKey: PropTypes.string,
};

LineChartComponent.defaultProps = {
  preview: false,
  xLabel: "",
  yLabel: "",
  xDataKey: "",
  yDataKey: "",
  lineDataKey: "",
};

export default LineChartComponent;
