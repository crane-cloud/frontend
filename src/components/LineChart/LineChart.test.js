import React from "react";
import { render } from "@testing-library/react";

import LineChart from ".";

describe("Test the line chart component", () => {
  const linechartData = [
    { name: "A", min: 400, max: 2400, sum: 2400 },
    { name: "B", min: 40, max: 240, sum: 240 },
  ];
  const linechartPropsNoPreview = {
    data: linechartData,
    preview: false,
    xLabel: "",
    yLabel: "",
    xDataKey: "",
    yDataKey: "",
    lineDataKey: "",
  };
  const linechartPropsPreivew = { ...linechartPropsNoPreview, preview: true };
  const linechartNoData = { ...linechartPropsNoPreview, data: [] };

  const LineChartNoPreview = render(<LineChart {...linechartPropsNoPreview} />);
  const LineChartPreview = render(<LineChart {...linechartPropsPreivew} />);
  const LineChartNoData = render(<LineChart {...linechartNoData} />);
  it("checks if the information bar component matches the snapshot", () => {
    expect(LineChartNoPreview).toMatchSnapshot();
  });
});
