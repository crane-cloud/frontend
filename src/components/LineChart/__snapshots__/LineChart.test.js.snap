// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the line chart component checks if the information bar component matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="recharts-responsive-container GraphSection"
        style="width: 100%; height: 100%;"
      >
        <div
          style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
        />
      </div>
    </div>
    <div>
      <div
        class="recharts-responsive-container GraphSection"
        style="width: 100%; height: 100%;"
      >
        <div
          style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
        />
      </div>
    </div>
    <div>
      <div
        class="recharts-responsive-container GraphSection"
        style="width: 100%; height: 100%;"
      >
        <div
          class="NoGraphData"
          height="0"
          width="0"
        >
          Graph Data Unavailable
        </div>
        <div
          style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="recharts-responsive-container GraphSection"
      style="width: 100%; height: 100%;"
    >
      <div
        style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
